tests/setup_test.go

```diff
@@ -3664,3 +3675,283 @@ func TestExportImport(t *testing.T) {
+			expectedRunResult:      rstypes.RunResultFailed,
+			expectedRunPhase:       rstypes.RunPhaseFinished,
+			expectedRunPhaseEvents: []rstypes.RunPhase{rstypes.RunPhaseQueued, rstypes.RunPhaseRunning, rstypes.RunPhaseRunning, rstypes.RunPhaseFinished},
+		},
+		{
+			name: "test run setup config error",
+			config: `
+				{
+				  runserror:
+				}
+				`,
+			expectedRunResult:      rstypes.RunResultUnknown,
+			expectedRunPhase:       rstypes.RunPhaseSetupError,
+			expectedRunPhaseEvents: []rstypes.RunPhase{rstypes.RunPhaseSetupError},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dir := t.TempDir()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			rsListeningAddress := setupRunStatusServer(t)
+			rsUrl := fmt.Sprintf("http://%s", rsListeningAddress)
+
+			go startRunStatusServer(t, ctx, rsListeningAddress)
+
+			sc := setup(ctx, t, dir, withGitea(true), withWebhooks(fmt.Sprintf("%s/%s", rsUrl, "runevents"), ""))
+			defer sc.stop()
+
+			giteaToken, tokenUser01 := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+
+			giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+			giteaClient := gitea.NewClient(giteaAPIURL, giteaToken)
+			gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, tokenUser01)
+
+			giteaRepo, project := createProject(ctx, t, giteaClient, gwClient)
+
+			push(t, tt.config, giteaRepo.CloneURL, giteaToken, "commit", false)
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+				if err != nil {
+					return false, nil
+				}
+
+				if len(runs) == 0 {
+					return false, nil
+				}
+				if runs[0].Phase != tt.expectedRunPhase {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			if len(runs) != 1 {
+				t.Fatalf("expected %d run got: %d", 1, len(runs))
+			}
+
+			run := runs[0]
+			if run.Phase != tt.expectedRunPhase {
+				t.Fatalf("expected run phase %q, got %q", tt.expectedRunPhase, run.Phase)
+			}
+			if run.Result != tt.expectedRunResult {
+				t.Fatalf("expected run result %q, got %q", tt.expectedRunResult, run.Result)
+			}
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				statuses, err := getRunEvents(t, rsUrl)
+				if err != nil {
+					return false, nil
+				}
+				if len(statuses) < len(tt.expectedRunPhaseEvents) {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			runPhaseEvents, err := getRunEvents(t, rsUrl)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			if diff := cmp.Diff(tt.expectedRunPhaseEvents, runPhaseEvents); diff != "" {
+				t.Fatalf("run phases mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+
+}
+
+func setupRunStatusServer(t *testing.T) string {
```

Comment:
Use a object to contain the status serve data, stop, start it like done for the other services and use a different file.